File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d = dict()
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            d[s[i]] = d.get(s[i], 0) + 1
            d[t[i]] = d.get(t[i], 0) - 1
            if d[t[i]] == 0:
                del d[t[i]]
            if s[i] in d and d[s[i]] == 0:
                del d[s[i]]
        return len(d) == 0
            
        
            

File: 242-valid-anagram/242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary <char, int> d = new Dictionary<char, int>();
        foreach (char c in s) {
            if (d.ContainsKey(c)) {
                d[c] += 1;
            } else {
                d[c] = 1;
            }
        }
        foreach (char c in t) {
            if (d.ContainsKey(c)) {
                d[c] -= 1;
            } else {
                return false;
            }
        }
        foreach (var (key, val) in d) {
            if (val != 0) {
                return false;
            }
        }
        return true;
    }
}